src/test/java/selenium/webdriver/DownloadScriptTest.java

```diff
@@ -0,0 +1,65 @@
+package selenium.webdriver;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import static java.time.Duration.ofSeconds;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated;
+
+public class DownloadScriptTest {
+
+    private static final String URL = "https://demo.seleniumeasy.com/bootstrap-download-progress-demo.html";
+
+    private WebDriver driver;
+
+    @BeforeEach
+    public void setup() {
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--ignore-ssl-errors=yes", "--ignore-certificate-errors");
+        driver = new ChromeDriver(options);
+    }
+
+    @Test
+    @DisplayName("Download and refresh the page when percentage is >= 50")
+    public void refreshPageTest() {
+        driver.get(URL);
+        driver.findElement(By.id("cricle-btn")).click();
+        assertTrue(isPercentGraterThenFifty(), "Percent is less than fifty");
+        driver.navigate().refresh();
+        String percentText = driver.findElement(By.cssSelector(".percenttext")).getText();
+        assertEquals("0%", percentText, "Percent is reloaded incorrectly");
+    }
+
+    @AfterEach
+    public void cleanup() {
+        driver.quit();
+    }
+
+    private boolean isPercentGraterThenFifty() {
+        new WebDriverWait(driver, ofSeconds(6)).until(visibilityOfElementLocated (By.cssSelector(".percenttext")));
+        boolean flag = false;
+        long start = System.currentTimeMillis();
+        long end = start + 30000;
+        while (System.currentTimeMillis() < end) {
+            WebElement percent = driver.findElement(By.cssSelector(".percenttext"));
+            String percentString = percent.getText().replace("%", "");
+            int percentNumber = Integer.parseInt(percentString);
+            if (percentNumber >= 50) {
+                flag = true;
+                break;
+            }
+        }
+        return flag;
```

Comment:
this method is good, but could be improved. We can discuss disadvantages of this logic

Reply:
The method is improved.